src/components/checkbox/checkbox.scss

```diff
@@ -0,0 +1,401 @@
+@import "default-theme";
+@import "theme-functions";
+@import "variables";
+
+/** The width/height of the checkbox element. */
+$md-checkbox-size: 18px !default;
+/** The width of the line used to draw the checkmark / mixedmark. */
+$md-checkbox-mark-stroke-size: 2/15 * $md-checkbox-size !default;
+/** The width of the checkbox border shown when the checkbox is unchecked. */
+$md-checkbox-border-width: 2px;
+/** The color of the checkbox border. */
+$md-checkbox-border-color: if($md-is-dark-theme, rgba(white, 0.7), rgba(black, 0.54)) !default;
+/** The color of the checkbox's checkmark / mixedmark. */
+$md-checkbox-mark-color: map-get($md-background, background);
+/** The color that is used as the checkbox background when it is checked. */
+$md-checkbox-background-color: md-color($md-primary, 500);
+/** The base duration used for the majority of transitions for the checkbox. */
+$md-checkbox-transition-duration: 140ms;
```

Comment:
AFAIK it's specific to the checkbox. My thinking behind variable-izing it was that if you had unique checkboxes where you wanted to change the overall durations of things (e.g. slow down the duration on a very large checkbox), this would allow you to do so. I'm happy to make it "private" if you'd prefer!